src/lib/descriptor/makefiles/rust.toml

```diff
@@ -733,17 +733,40 @@ run_task = "do-copy-apidocs"
 description = "Copies the generated documentation to the docs/api directory."
 category = "Documentation"
 private = true
-script_runner = "@shell"
+script_runner = "@duckscript"
 script = [
 '''
-SOURCE_DIRECTORY=${CARGO_MAKE_DOCS_ROOT_FOLDER}/target/doc
-TARGET_DIRECTORY=${CARGO_MAKE_DOCS_ROOT_FOLDER}/docs/api/${CARGO_MAKE_DOCS_SUB_FOLDER}
+if is_empty ${CARGO_MAKE_DOCS_SUB_FOLDER}
+    SOURCE_DIRECTORY= set ${CARGO_MAKE_DOCS_ROOT_FOLDER}
+else
+    SOURCE_DIRECTORY= set ${CARGO_MAKE_DOCS_ROOT_FOLDER}/${CARGO_MAKE_DOCS_SUB_FOLDER}
+end
+
+if is_dir ${SOURCE_DIRECTORY}/target/${CARGO_MAKE_CRATE_TARGET_TRIPLE}/doc
+    SOURCE_DIRECTORY= set ${SOURCE_DIRECTORY}/target/${CARGO_MAKE_CRATE_TARGET_TRIPLE}/doc
+else
+    SOURCE_DIRECTORY= set ${SOURCE_DIRECTORY}/target/doc
+end
+
+if is_empty ${CARGO_MAKE_DOCS_SUB_FOLDER}
```

Comment:
just in case to make sure CARGO_MAKE_DOCS_SUB_FOLDER is defined, let change to:
if is_empty "${CARGO_MAKE_DOCS_SUB_FOLDER}"

Reply:
yes